experiment/cherrypicker/server.go

```diff
@@ -0,0 +1,399 @@
+			continue
+		}
+		// TODO: Collect all "/cherrypick" comments and figure out if any
+		// comes from an org member?
+		targetBranch = cherryPickMatches[0][1]
+		commentAuthor = c.User.Login
+		break
+	}
+	if targetBranch == "" || commentAuthor == "" {
+		return nil
+	}
+	return s.handle(org, repo, commentAuthor, baseBranch, targetBranch, num)
+}
+
+var cherryPickBranchFmt = "cherry-pick-%d-to-%s"
+
+func (s *Server) handle(org, repo, commentAuthor, baseBranch, targetBranch string, num int) error {
+	// TODO: Use a whitelist for allowed base and target branches.
+	if baseBranch == targetBranch {
+		resp := fmt.Sprintf("Base branch (%s) needs to differ from target branch (%s)", baseBranch, targetBranch)
+		s.log.Info(resp)
+		return s.ghc.CreateComment(org, repo, num, resp)
+	}
+
+	// Only members should be able to do cherry-picks.
+	ok, err := s.ghc.IsMember(org, commentAuthor)
+	if err != nil {
+		return err
+	}
+	if !ok {
+		resp := fmt.Sprintf("Only [%s](https://github.com/orgs/%s/people) org members may request cherry picks. You can still do the cherry-pick manually.", org, org)
+		s.log.Info(resp)
+		return s.ghc.CreateComment(org, repo, num, resp)
+	}
+
+	if err := s.ensureForkExists(org, repo); err != nil {
+		return err
+	}
+
+	// Clone the repo, checkout the target branch.
+	startClone := time.Now()
+	r, err := s.gc.Clone(org + "/" + repo)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := r.Clean(); err != nil {
+			s.log.WithError(err).Error("Error cleaning up repo.")
+		}
+	}()
+	if err := r.Checkout(targetBranch); err != nil {
+		return err
+	}
+	s.log.WithField("duration", time.Since(startClone)).Info("Cloned and checked out target branch.")
+
+	// Fetch the patch from Github
+	localPath, err := s.getPatch(org, repo, num)
+	if err != nil {
+		return err
+	}
+
+	if err := r.Config("user.name", "cherrypicker"); err != nil {
+		return err
+	}
+	if err := r.Config("user.email", "cherrypicker@localhost"); err != nil {
+		return err
+	}
+
+	// Checkout a new branch for the cherry-pick.
+	newBranch := fmt.Sprintf(cherryPickBranchFmt, num, targetBranch)
+	if err := r.CheckoutNewBranch(newBranch); err != nil {
+		return err
+	}
+
+	// Apply the patch.
+	if err := r.Apply(localPath); err != nil {
+		resp := fmt.Sprintf("PR %d failed to apply on top of branch %q: %v", num, targetBranch, err)
+		return s.ghc.CreateComment(org, repo, num, resp)
+	}
+
+	// Push the new branch in the bot's fork.
+	if s.push == nil {
+		s.push = r.Push
+	}
+	if err := s.push(s.botName, s.credentials, repo, newBranch); err != nil {
+		resp := fmt.Sprintf("Failed to push cherry-picked changes in Github: %v", err)
+		return s.ghc.CreateComment(org, repo, num, resp)
+	}
+
+	// Open a PR in Github.
+	title := fmt.Sprintf("Automated cherry-pick of #%d on %s", num, targetBranch)
+	// TODO: Make this a configurable template
+	body := fmt.Sprintf("This is an automated cherry-pick of #%d\n\n/assign %s", num, commentAuthor)
```

Comment:
Change this to `@`-mention users to trigger github notifications.
`/assign @%s`

Reply:
Is there any actual gain? Github already sends an email when I am assinged somewhere.

Reply:
In that case there isn't much gain. I guess using `@` also makes the username a hyperlink. Either way is fine though.